### PR TITLE
Add support for extra small (min 16 byte) pages to Heimdall.

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

File: heimdall/heim_main.c [hunk 1]
```diff
@@ -210,12 +210,12 @@ static void heim_fini(Int exitcode)
...
     {
-      long size = (long)(current_page_ptr->start_addr + clo_page_size - first_page_ptr->start_addr) / 1024;
+      long size = (long)(current_page_ptr->start_addr + clo_page_size - first_page_ptr->start_addr);
 
-      VG_(printf)("  Modified pages: 0x%lx - 0x%lx [size: %ld Kb]\n",
+      VG_(printf)("  Modified area: 0x%lx - 0x%lx size: %ld bytes [%ld.%ld Kb]\n",
                   (long)first_page_ptr->start_addr,
                   (long)(current_page_ptr->start_addr + clo_page_size),
-                  size);
+                  size, size / 1024, ((size % 1024) * 10) / 1024);
```
Comment (elecro) [inline heimdall/heim_main.c]: We should put each of the arguments in a new line, the other args are already on their own line.

File: heimdall/heim_main.c [hunk 2]
```diff
@@ -224,12 +224,13 @@ static void heim_fini(Int exitcode)
     current_page_ptr = current_page_ptr->ordered_next;
   }
 
-  VG_(printf)("Total modified pages: %ld Kb\n", total_size);
+  VG_(printf)("Total modified memory area: %ld bytes [%ld.%ld Kb]\n",
+              total_size, total_size / 1024, ((total_size % 1024) * 10) / 1024);
```
Comment (elecro) [inline heimdall/heim_main.c]: ditto.

File: heimdall/heim_main.c [hunk 3]
```diff
@@ -244,7 +245,7 @@ static Bool heim_process_cmd_line_option(const HChar* arg)
 static void heim_print_usage(void)
 {
    VG_(printf) (
-"    --page_size=<number>            page size [4096]\n"
+"    --page_size=<number>            page size [%d]\n", clo_page_size
```
Comment (elecro) [inline heimdall/heim_main.c]: ditto

**Comment by zherczeg** [comment]
Updated.

**Comment by galpeter** [comment]
lgtm

**Comment by elecro** [comment]
lgtm

**Pull request merged**